src/main/kotlin/org/rust/lang/core/completion/RsVisRestrictionCompletionProvider.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.lang.core.completion
+
+import com.intellij.codeInsight.completion.CompletionParameters
+import com.intellij.codeInsight.completion.CompletionResultSet
+import com.intellij.codeInsight.lookup.LookupElementBuilder
+import com.intellij.patterns.PlatformPatterns.psiElement
+import com.intellij.patterns.PsiElementPattern
+import com.intellij.psi.PsiElement
+import com.intellij.util.ProcessingContext
+import org.rust.ide.icons.RsIcons
+import org.rust.lang.core.psi.RsElementTypes
+import org.rust.lang.core.psi.RsPath
+import org.rust.lang.core.psi.RsVisRestriction
+import org.rust.lang.core.psi.ext.qualifier
+import org.rust.lang.core.psiElement
+import org.rust.lang.core.with
+
+/**
+ * Provides completion inside visibility restriction:
+ * `pub(<here>)`
+ */
+object RsVisRestrictionCompletionProvider : RsCompletionProvider() {
+    override val elementPattern: PsiElementPattern.Capture<PsiElement>
+        get() = psiElement(RsElementTypes.IDENTIFIER)
+            .withParent(
+                psiElement<RsPath>()
+                    .with("hasOneSegment") { item, _ -> item.qualifier == null && item.path == null && !item.hasColonColon }
```

Comment:
`item.qualifier` and `item.path` for are the same. Maybe you want `typeQual`?

Reply:
Actually, I wanted to check if the path has no `::<sub-path>` on the "right" side, but now I realized that due to the way paths are parsed, this is not relevant here. And `typeQual` was missed in the check, so thanks :D